File: RetrieveJSON/app/src/main/java/mx/itesm/retrievejson/MainActivity.java
package mx.itesm.retrievejson;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.Toast;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class MainActivity extends AppCompatActivity implements RetrieveJson.JSONListener, AdapterView.OnItemClickListener{

    public ListView list_hobbies;
    private JSONArray json_data;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        this.list_hobbies = (ListView) findViewById(R.id.friends);
        this.list_hobbies.setOnItemClickListener(this);
    }

    public void LoadList(View view) {
        RetrieveJson retrieveJson = new RetrieveJson(this);
        retrieveJson.execute("https://raw.githubusercontent.com/MarcoRob/marcorob.github.io/master/friends.json");

        /*JSONAdapter adpater = new JSONAdapter(this.json_data, this);
        this.list_hobbies.setAdapter(adpater);*/
    }

    @Override
    public void requestDone(JSONArray jsonArray) {
        try {
            this.json_data = jsonArray;
            JSONAdapter adapter = new JSONAdapter(jsonArray, this);
            this.list_hobbies.setAdapter(adapter);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
        //Toast.makeText(this, i+" pos", Toast.LENGTH_SHORT).show();
        Intent intent = new Intent(this, JSONDetail.class);
        try {
            Log.d("JSON POS " + i, this.json_data.getJSONObject(i).toString());
            intent.putExtra("json", this.json_data.getJSONObject(i).toString());
            startActivity(intent);
        } catch (Exception e) {
            Toast.makeText(this, "Try Later", Toast.LENGTH_SHORT).show();
        }
    }

}

File: Activity31_Kotlin/app/src/main/java/mx/itesm/activity31_kotlin/MainActivity.kt
package mx.itesm.activity31_kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ListView
import android.widget.Toast
import org.json.JSONArray

class MainActivity : AppCompatActivity(), AdapterView.OnItemClickListener {

    var foodMenu : ListView? = null;
    var foodOptions : JSONArray = JSONArray("[{'name':'burger', 'price':15, 'description':'a juicy burger!'}," +
                                                "{'name':'hotdog', 'price':20, 'description':'an ok hot dog'}," +
                                                "{'name':'tacos', 'price':12, 'description':'some pretty good tacos'}," +
                                                "{'name':'torta', 'price':22, 'description':'nice torta'}," +
                                                "{'name':'carne asada', 'price':50, 'description':'a great carne asada'}]")


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        this.foodMenu =  findViewById(R.id.foodMenu);
        var customAdapter : CustomAdapter = CustomAdapter(foodOptions, this);

        foodMenu?.adapter = customAdapter;
        foodMenu?.setOnItemClickListener(this);
    }

    override fun onItemClick(p0: AdapterView<*>?, p1: View?, p2: Int, p3: Long) {
        var description : String = foodOptions.getJSONObject(p2).getString("description").toString();
        Toast.makeText(this, description, Toast.LENGTH_SHORT).show();
    }


}
